crates/rome_js_formatter/src/utils/test_each_pattern.rs

```diff
@@ -0,0 +1,294 @@
+use crate::js::expressions::template_element::TemplateElementOptions;
+use crate::js::lists::template_element_list::{TemplateElementIndention, TemplateElementLayout};
+use crate::prelude::*;
+use rome_formatter::printer::Printer;
+use rome_formatter::{
+    format_args, write, CstFormatContext, FormatOptions, RemoveSoftLinesBuffer, VecBuffer,
+};
+use rome_js_syntax::{JsAnyTemplateElement, JsTemplateElementList};
+use rome_text_size::TextRange;
+use std::cmp;
+use unicode_width::UnicodeWidthStr;
+
+#[derive(Debug)]
+enum EachElement {
+    /// A significant value in the test each table. It's a row element.
+    Column(EachColumn),
+    /// Indicates the end of the current row.
+    LineBreak,
+}
+
+/// Row element containing the column information.
+#[derive(Debug)]
+struct EachColumn {
+    /// Formatted text of the column.
+    text: String,
+    /// Corresponding range for the text to replace it.
+    range: TextRange,
+    /// Indicates the line break in the text.
+    will_break: bool,
+}
+
+#[derive(Debug)]
+pub(crate) struct EachTable {
+    /// Holds information about the current row.
+    current_row: EachCurrentRow,
+    /// Information about all rows.
+    rows: Vec<EachRow>,
+    /// Contains the maximum length of each column of all rows.
+    columns_width: Vec<usize>,
+    /// Elements for formatting.
+    elements: Vec<EachElement>,
+}
+
+#[derive(Debug)]
+struct EachCurrentRow {
+    /// Contains the maximum length of the current column.
+    columns_width: Vec<usize>,
```

Comment:
Nit: Use `TextSize` instead of `usize`

```suggestion
    column_widths: Vec<usize>,
```

`columns_width` -> `column_widths` to make it clear its the widths of the columns and not the total width of all columsn